internal/services/springcloud/spring_cloud_java_deployment_resource.go

```diff
@@ -356,31 +356,17 @@ func resourceSprintCloudJavaDeploymentSchema() map[string]*pluginsdk.Schema {
 						Type:     pluginsdk.TypeString,
 						Optional: true,
 						Computed: true,
-						ValidateFunc: validation.StringInSlice([]string{
-							"500m",
-							"1",
-							"2",
-							"3",
-							"4",
-						}, false),
+						// There are additional options when the service team allow them
+						ValidateFunc: validation.StringIsNotEmpty,
 					},
 
 					// The value returned in GET will be recalculated by the service if the deprecated "memory_in_gb" is honored, so make this property as Computed.
 					"memory": {
 						Type:     pluginsdk.TypeString,
 						Optional: true,
 						Computed: true,
-						ValidateFunc: validation.StringInSlice([]string{
-							"512Mi",
-							"1Gi",
-							"2Gi",
-							"3Gi",
-							"4Gi",
-							"5Gi",
-							"6Gi",
-							"7Gi",
-							"8Gi",
-						}, false),
+						// There are additional options when the service team allow them
```

Comment:
```suggestion
							// NOTE: we're intentionally not validating this field since additional values are possible when enabled by the service team
```